### ISSUE TITLE
Hotkey: Clicking on non-transparent pixels select Object.

### ISSUE THREAD

**Issue by UliAbo** [mapeditor/tiled#3330]
### Purpose
I have lots of rather round Tiles (Objects like Items) in my single ObjectLayer and selecting them - especially when they partially overlap - is quite tedious. There is often quite some transparent space within one sprite.

### Describe the solution you'd like
While pressing hotkey "tbd", I select Objects only when I click on their actual sprite pixels and transparent pixels are ignored.

### Describe alternatives you've considered
"Hold Alt and right click and then you'll get a drop-down menu with all the objects in that spot, so you can access the overlapped objects." --> That's useful, but requires extra clicks and needs extra thinking to translate the desired visual Object into the Object name (if given), searching through the list then. That takes some moments in my brain to process. Not so intuitive for me.

**Comment by eishiya** [comment]
I wonder if rather than a hotkey, which also requires extra thinking, much like the current alt+right click method, this could be an object selection toggle in Preferences.

**Comment by UliAbo** [comment]
True! I like your suggestion better. You're right - if such a preference would be available, I would definitely enable it - because I don't think I see a use case for myself, where I DO want to select Objects through their transparent pixels if I could have it the other way like suggested... Hmm.

**Comment by eishiya** [comment]
> I don't think I see a use case for myself, where I DO want to select Objects through their transparent pixels

It could be useful in scenarios where you have tilesets with large tiles but very variable-size objects and are zoomed out. In those cases, I think it could be useful if you could click on the transparent space of an object, as the non-transparent area would be so small. So, tiles that are less like your tableware, and more like your chairs, with lots of empty space. However, if I were using a tileset where that's a common problem, I'd probably leave that "ignore transparent Tile Object pixels when selecting" setting off.

**Comment by UliAbo** [comment]
That makes sense!

I would like to add some additional info to this issue, to better understand my situation: 
![image](https://user-images.githubusercontent.com/1249266/163317776-4a8c336f-17ce-4a91-94a3-35bd7b7ac509.png)

Just some info of the current situation:
- I have a highly interactive game world with lots of single draggable small Items (like above).
- To design and improve this world in Tiled, I drag around small sprites all the time as you can imagine.
- The pain is that currently Tiled kind of forces me to use "collection of images" to use single PNG files for all the items which have individual image measurements and are only as large as they need to be.

To handle and organize these tons of small sprites I would normally use sprite sheets. Sprite sheets are not only more performant for the game engine but are also easier to organize because I don't need to give file names for all of the single items. And are also easier to edit in image software instead of single files.

So for my way of designing the world in Tiled with that many items, it's unfortunately currently not possible to use sprite sheets with a standardized size (tiles) because there would be even more transparent space. Using those standardized tiles for our items would make it even more tedious to design the world because there would be so much overlapping of transparent space and selecting would be a nightmare.

Implementing this desired feature with "ignore transparent Tile Object pixels when selecting" would help greatly because then I could go back to using a sprite sheet and all transparent space would be no issue. Then I could directly select what I see in Tiled and this would be then the same WYSIWYG effect like I have ingame in my project.
